plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceRuntimeInfo.java

```diff
@@ -310,16 +316,33 @@ public String projectsRoot() {
     protected Map<String, ServerImpl> getServersWithFilledPorts(final String externalHostame, final String internalHostname, final Map<String, List<PortBinding>> exposedPorts) {
         final HashMap<String, ServerImpl> servers = new LinkedHashMap<>();
 
+        boolean useMappedPorts = true;
+        if (useInternalContainerAddresses()) {
+            if (info.getNetworkSettings() != null && internalHostname.equals(info.getNetworkSettings().getIpAddress())) {
+                useMappedPorts = false;
```

Comment:
You're right, that was the goal. I'll remove this check and rename the variable for clarity.